src/types.py

```diff
@@ -995,10 +1029,12 @@ def demangled_type(sym_type: CxxType, sym_name: Optional[CxxTerm] = None) -> "Ty
                     and len(sym_name.qualified_name) > 1
                     and final_name not in CxxSymbol.STATIC_FUNCTIONS
                 ):
+                    class_type = type_for_class(sym_name.qualified_name[:-1])
                     # NB: This assumes `this` is passed as the first arg, which is true
                     # for most thiscall methods on PPC. However, this is incorrect for
                     # static member functions, functions returning structs, and other ABIs.
-                    params.append(FunctionParam(type=Type.ptr(), name="this"))
+                    params.append(FunctionParam(type=Type.ptr(class_type), name="this"))
+                    # TODO: Virtual methods may take a second argument here
```

Comment:
what second argument? this comment is going to confuse me in the future

Reply:
I'm now confident that this was a mistake to begin with. The issue has nothing to do with virtual methods, I think it is just that when the function returns a struct/class, then there's an argument with a pointer to that return value (and that argument nor the return value is indicated in the mangled name).